tracecontext.go

```diff
@@ -169,23 +169,61 @@ type TraceState struct {
 
 // NewTraceState returns a TraceState based on entries.
 func NewTraceState(entries ...TraceStateEntry) TraceState {
-	out := TraceState{}
+	var out TraceState
 	var last *TraceStateEntry
+	var modified []TraceStateEntry
+	recorded := make(map[string]uint8)
+	// Range over the entries and find duplicate keys. When a previous key is
+	// found, it is stored in a map with a count that is incremented every time
+	// the key is observed.
+	// When keys are observed more than once, the last value observed is stored
+	// in a slice of the modified keys.
 	for _, e := range entries {
-		e := e // copy
+		if count, ok := recorded[e.Key]; ok {
+			if count > 1 {
+				for i, m := range modified {
+					if m.Key == e.Key {
+						modified[i] = e
```

Comment:
Once we've found a matching key, we could `continue`, right? Since we're only wanting to record the latest entry `e` for a specific key `e.Key`?

Reply:
yeah we could if we kept the implementation as is, I'm going to simplify it and change it to only act on the 'es' vendor key which is the only one we care about, it'll also eliminate allocations :)